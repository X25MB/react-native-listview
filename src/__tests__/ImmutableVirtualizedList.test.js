import { data, expectors } from './testUtils';

describe('ImmutableListView', () => {
  it('renders with empty data', () => {
    expectors.expectVirtualizedToMatchSnapshotWithData(data.EMPTY_DATA);
  });

  it('renders basic List', () => {
    expectors.expectVirtualizedToMatchSnapshotWithData(data.LIST_DATA);
  });

  it('renders nested List', () => {
    expectors.expectVirtualizedToMatchSnapshotWithData(data.LIST_DATA_NESTED);
  });

  it('renders basic Range', () => {
    expectors.expectVirtualizedToMatchSnapshotWithData(data.RANGE_DATA);
  });
});
