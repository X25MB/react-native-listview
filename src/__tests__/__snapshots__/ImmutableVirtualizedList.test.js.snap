// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImmutableVirtualizedList renders basic List 1`] = `
<RCTScrollView
  data={
    Array [
      "lists",
      "are",
      "great",
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  immutableData={
    Array [
      "lists",
      "are",
      "great",
    ]
  }
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  shouldItemUpdate={[Function]}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        "lists"
      </Text>
    </View>
    <View
      onLayout={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        "are"
      </Text>
    </View>
    <View
      onLayout={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        "great"
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`ImmutableVirtualizedList renders basic Range 1`] = `
<RCTScrollView
  data={
    Array [
      3,
      6,
      9,
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  immutableData={
    Array [
      3,
      6,
      9,
    ]
  }
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  shouldItemUpdate={[Function]}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        3
      </Text>
    </View>
    <View
      onLayout={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        6
      </Text>
    </View>
    <View
      onLayout={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        9
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`ImmutableVirtualizedList renders nested List 1`] = `
<RCTScrollView
  data={
    Array [
      Array [
        "so",
        "are",
      ],
      Array [
        "nested",
        "lists",
      ],
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  immutableData={
    Array [
      Array [
        "so",
        "are",
      ],
      Array [
        "nested",
        "lists",
      ],
    ]
  }
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  shouldItemUpdate={[Function]}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        ["so","are"]
      </Text>
    </View>
    <View
      onLayout={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        ["nested","lists"]
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`ImmutableVirtualizedList renders with empty data 1`] = `
<RCTScrollView
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  immutableData={Array []}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  shouldItemUpdate={[Function]}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
